server/service/src/sync/translations/master_list.rs

```diff
@@ -59,6 +132,92 @@ impl SyncTranslation for MasterListTranslation {
     }
 }
 
+#[derive(Clone)]
+struct GenerateRequisitionProgram {
+    pub program_row: ProgramRow,
+    pub program_requisition_settings_row: ProgramRequisitionSettingsRow,
+    pub program_requisition_order_type_rows: Vec<ProgramRequisitionOrderTypeRow>,
+}
+
+fn generate_requisition_program(
+    connection: &StorageConnection,
+    master_list: LegacyListMasterRow,
+) -> Result<Option<GenerateRequisitionProgram>, anyhow::Error> {
+    if master_list.is_program == false {
+        return Ok(None);
+    }
+
+    let program_settings = master_list.program_settings.ok_or(anyhow::anyhow!(
+        "Program settings not found for program {}",
+        master_list.id
+    ))?;
+    let order_types = program_settings
+        .store_tags
+        .tags
+        .values()
+        .filter_map(|tag| tag.order_types.clone())
```

Comment:
We would need to iterate over tags and insert program_settings for each tag, and then link order_types to those program_settings